en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/troubleshooting/debug-memory.md

```diff
@@ -0,0 +1,39 @@
+# Taking Memory Dumps
+
+As with any software product, there can be instances where WSO2 Choreo Connect cluster fails due to a memory resource exhaustion. The heap dumps will always point you towards the cause of the memory leak. Therefore, it is important to be able to retrieve heap dumps from an environment at the point when an error occurs. This will avoid the necessity of reproducing the exact issue again (specially, in the case of production issues). A resource exhaustion can happen for two reasons:
+
+- Due to a bug in the system.
+- An actual limitation of resources based on low configuration values.
+
+Following sections will guide you on how to take heap dumps from each component of WSO2 Choreo Connect.
```

Comment:
We have mentioned only Adapter and Enforcer. Shall we specify that it's not applicable for Router if that's the case?

Reply:
Its applicable for router too. However the steps are not finalized yet, since we are planning the approach we are taking to get router's dump. This will be available before the 1.0.0-GA